File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        
        def traversal(curr):
            if not curr:
                return None
            
            temp = traversal(curr.left)
            curr.left = traversal(curr.right)
            curr.right = temp
            
            return curr
        
        return traversal(root)
                
        

File: 0226-invert-binary-tree/0226-invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    
    if(root === null) return null;
    if(root.left !== null && root.right !== null){
        const temp = root.left;
        root.left = root.right;
        root.right = temp;
    } 
    else if(root.left === null){
        root.left = root.right;
        root.right = null;
    }
    else{
        root.right = root.left;
        root.left = null;
    }
    
    invertTree(root.left);
    invertTree(root.right);
    
    return root;
    
};